docs/source/model_design/vivarium_model_components/risk_factors/risk_effects/index.rst

```diff
@@ -102,38 +102,76 @@ as python code:
-        f_rr = scipy.interpolate.interp1d(x, y, kind='linear', bounds_error=False, fill_value=(min(y), max(y)))
-
-        # pick a tmrel between tmred.min and tmred.max --- for certain risk factors, the modeling team uploads a model for this, too --- shiny tool has information about this
-        # and calculate relative risk at tmrel
-        tmrel = np.random.uniform(tmred.min, tmred.max)
-        rr_at_tmrel = f_rr(tmrel)
-        t4 = t3 / rr_at_tmrel
-
-        # TODO: see if GBD would also benefit from clipping these RRs
-        t5 = np.clip(t4, 1.0, np.inf)
-
-        y = np.array(t5.values)
-        f_rr_2 = scipy.interpolate.interp1d(x, y, kind='linear', bounds_error=False, fill_value=(min(y), max(y)))
-        return f_rr_2
+  import numpy as np
+  import scipy.interpolate
+  import matplotlib.pyplot as plt
+  import gbd_mapping, vivarium_gbd_access
+
+  # Replace with your risk of interest
+  risk = gbd_mapping.risk_factors.high_systolic_blood_pressure # high_body_mass_index_in_adults
+  # Replace with your cause of interest
+  cause = gbd_mapping.causes.ischemic_heart_disease
+  age_group_id = 20 # 75 to 79
+  sex_id = 1 # Male
+  year_id = 2021
+
+  relative_risk_data = vivarium_gbd_access.gbd.get_relative_risk(
+      risk.gbd_id,
+      1, # Global
+      year_id=year_id,
+  )
+
+  # Subset to cause, age, and sex of interest
+  # If interested in multiple, would loop through them
+  relative_risk_data = relative_risk_data[
+      (relative_risk_data.cause_id == cause.gbd_id) &
+      (relative_risk_data.age_group_id == age_group_id) &
+      (relative_risk_data.sex_id == sex_id)
+  ].sort_values('exposure')
+
+  relative_risk_functions = {}
+
+  # Do calculation at the draw level
+  for draw_id in range(1_000):
+    relative_risk_draw = relative_risk_data[f'draw_{draw_id}']
+    # interpolate a continuous function between the points,
+    # and extrapolate outside the range with the endpoints
+    raw_relative_risk_function = scipy.interpolate.interp1d(
+        relative_risk_data.exposure,
+        relative_risk_draw,
+        kind='linear',
+        bounds_error=False,
+        fill_value=(
+            relative_risk_draw.min(),
+            relative_risk_draw.max(),
+        )
+    )
+
+    # pick a tmrel between tmred.min and tmred.max and calculate relative risk at tmrel
+    # for certain risk factors, the modeling team uploads a model for this with TMREL draws --
+    # those should be used instead of this, when available!
+    tmrel = np.random.uniform(risk.tmred.min, risk.tmred.max)
+    rr_at_tmrel = raw_relative_risk_function(tmrel)
+    normalized_relative_risk_draw = relative_risk_draw / rr_at_tmrel
+
+    # Clip relative risk to 1, since the TMREL is supposed to represent minimum risk
+    # NOTE: It is not clear if this is always appropriate, since TMRELs are across all
+    # risk-cause pairs involving this risk, not only the pair we are modeling
+    # TODO: see if GBD would also benefit from clipping these RRs
```

Comment:
I'm not sure what this comment means... But along with all the other discussion, it makes me think the answer to my question in the doc session ("Is it unclear how GBD computes the PAF?") should have been "Yes." Let me make my question more precise:

1. Does GBD clip the RRs to 1 when computing the PAF? (My understanding is that, based on the code, it looks like the answer is "no.")

2. When we clip to 1, do we get the same PAF as GBD?

   - If the answer is "yes", then why are we getting the same result as GBD with a different RR curve (assuming the answer to Question 1 is "no")?
   - If the answer is "no", then we should note that somewhere (but it seems like the answer is "yes" since Abie's goal was to match GBD's PAF).

3. If we don't clip to 1, do we know for sure that we'd get a negative PAF sometimes, or is that just a hypothetical possibility?

Maybe I'm misunderstanding something, but it seemed like we still had some outstanding questions about this issue.

With regard to this PR, is it possible to clarify what this comment means?

Reply:
Yes I see your point -- I don't think it is totally clear what GBD did. IIUC @aflaxman's conclusion was that he got similar numbers to GBD when he _did_ clip, but not when he didn't? That is confusing if we don't see GBD doing clipping.

Reply:
> Ok, looking back at [the Slack thread](https://ihme.slack.com/archives/CQ8MGENNA/p1717510424955569?thread_ts=1717448354.571769&cid=CQ8MGENNA) it looks like we matched GBD even without clipping. Given that and the conceptual problems with clipping, I'm not sure I see any reason to clip.

The PAF of high sodium on stomach cancer matches up better with clipping --- the median is negative without clipping, but positive in GBD.

Reply:
Ok let me take a look at that. I wonder if it has to do with how sodium is "weird"?

I can confirm that GBD does not clip, at least not uniformly, because it does estimate negative PAFs:

```python
import pandas as pd
import gbd_mapping, vivarium_gbd_access.gbd
sudan_alcohol_pafs = vivarium_gbd_access.gbd.get_paf(gbd_mapping.risk_factors.high_alcohol_use.gbd_id, 522, 2021)
pd.Series(
    sudan_alcohol_pafs[(sudan_alcohol_pafs.sex_id == 2) & (sudan_alcohol_pafs.cause_id == gbd_mapping.causes.ischemic_heart_disease.gbd_id)]
        .filter(like='draw').values.reshape(-1)
).hist(bins=100, log=True)
```

![image](https://github.com/ihmeuw/vivarium_research/assets/13357648/3b856c7f-2b9e-4223-9a82-442cb5ece62f)

Reply:
Is this showing a PAF of -350% for one draw??

Reply:
I'd assume -3.5%, though I'm not sure exactly what the units are.

Reply:
For posterity: answers to both questions 1 & 2 were "yes."